core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/MetricsServiceImpl.java

```diff
@@ -655,7 +656,16 @@ public void setDefaultTTL(int defaultTTL) {
                         .doOnNext(i -> insertedDataPointEvents.onNext(metric)))
                 .doOnNext(meter::mark);
 
-        Observable<Integer> indexUpdates = dataAccess.updateMetricsIndex(metrics)
+        Observable<Metric<T>> updateIdxMetrics = metrics.filter(metric -> {
```

Comment:
@jsanda, I only partially understand your last post with regards to what needs to be done. I would like to keep this PR simple and then enhance. I will clone the current ticket and emphasize to add the improvement to make it out-of-band. And we can tackle that in the next release.

To have the index updated out-of-band we will have to re-write the entire testing infrastructure, not just for the core services but also for REST. And I am more concerned about the impact on the REST integration tests to be honest.